hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/embedded/EmbeddedHedera.java

```diff
@@ -16,551 +16,74 @@
-import com.swirlds.platform.state.PlatformState;
-import com.swirlds.platform.system.Platform;
-import com.swirlds.platform.system.Round;
-import com.swirlds.platform.system.SoftwareVersion;
-import com.swirlds.platform.system.SwirldState;
-import com.swirlds.platform.system.address.Address;
-import com.swirlds.platform.system.address.AddressBook;
-import com.swirlds.platform.system.events.ConsensusEvent;
-import com.swirlds.platform.system.events.Event;
-import com.swirlds.platform.system.transaction.ConsensusTransaction;
-import com.swirlds.platform.system.transaction.SwirldTransaction;
-import com.swirlds.platform.test.fixtures.addressbook.RandomAddressBookBuilder;
 import edu.umd.cs.findbugs.annotations.NonNull;
-import edu.umd.cs.findbugs.annotations.Nullable;
-import java.io.IOException;
-import java.io.UncheckedIOException;
-import java.nio.file.Path;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Random;
-import java.util.concurrent.ArrayBlockingQueue;
-import java.util.concurrent.BlockingQueue;
-import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.stream.IntStream;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-
-/**
- * An embedded Hedera node that can be used in tests.
- */
-public class EmbeddedHedera {
-    private static final Logger LOGGER = LogManager.getLogger(EmbeddedHedera.class);
 
-    private static final int MAX_PLATFORM_TXN_SIZE = 1024 * 6;
-    private static final int MAX_QUERY_RESPONSE_SIZE = 1024 * 1024 * 2;
-    private static final Signature TOY_SIGNATURE = new Signature(SignatureType.RSA, new byte[384]);
-    private static final TransactionResponse OK_RESPONSE = TransactionResponse.getDefaultInstance();
-    private static final PlatformStatusChangeNotification ACTIVE_NOTIFICATION =
-            new PlatformStatusChangeNotification(ACTIVE);
-    private static final PlatformStatusChangeNotification FREEZE_COMPLETE_NOTIFICATION =
-            new PlatformStatusChangeNotification(FREEZE_COMPLETE);
-
-    private final Hedera hedera;
-    private final NodeId defaultNodeId;
-    private final AccountID defaultNodeAccountId;
-    private final AddressBook addressBook;
-    private final ToyPlatform platform;
-    private final PlatformState platformState = new PlatformState();
-    private final Map<AccountID, NodeId> nodeIds;
-    private final Map<NodeId, com.hedera.hapi.node.base.AccountID> accountIds;
-    private final FakeHederaState state = new FakeHederaState();
-    private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+public interface EmbeddedHedera {
```

Comment:
We should probably add more documentation when this is used and how it is different from regular tests. Need not be done in this PR. But will be helpful.

Reply:
We could also specify the tests that cannot use Embedded mode like the ones using ECDSA keys

Reply:
True Neeha, created [this issue](https://github.com/hashgraph/hedera-services/issues/14046) to add a detailed _package-info.java_ that covers the expected purpose for each `HederaNetwork` implementation.